backend/src/data/repositories/task-note/task-note.repository.ts

```diff
@@ -0,0 +1,57 @@
+import { Page } from 'objection';
+
+import {
+  EntityPagination,
+  TaskNoteCreateArgumentsDto,
+  TaskNoteGetAllArgumentsDto,
+  TaskNoteGetItemResponseDto,
+} from '~/common/types/types';
+import { TaskNote as TaskNoteM } from '~/data/models/models';
+
+type Constructor = {
+  TaskNoteModel: typeof TaskNoteM;
+};
+
+class TaskNote {
+  #TaskNoteModel: typeof TaskNoteM;
+
+  public constructor({ TaskNoteModel }: Constructor) {
+    this.#TaskNoteModel = TaskNoteModel;
+  }
+
+  public async getAll({
+    count,
+    page,
+    taskId,
+  }: TaskNoteGetAllArgumentsDto): Promise<
+    EntityPagination<TaskNoteGetItemResponseDto>
+  > {
+    const { results, total } = await this.#TaskNoteModel
+      .query()
+      .where({ taskId })
+      .withGraphFetched('author(withoutPassword).[userDetails]')
+      .page(page, count)
+      .castTo<Page<TaskNoteM & TaskNoteGetItemResponseDto>>();
```

Comment:
Bitwise operator here as well?

Reply:
`.page` return the data in Model format (`TaskNoteM`), but I also use `.withGraphFetched`, so I have to cast it to `TaskNoteGetItemResponseDto`. The problem with `Page<TaskNoteGetItemResponseDto>` is that `.page` is expected to return all the other properties of model as well, not only our data, so we cast the data to the type, which has all the model properties (`TaskModelM`) and to the type, which has the properties we need (`TaskNoteGetItemResponseDto`)

Reply:
Got it! Thank you